tabular/src/autogluon/tabular/models/knn/knn_model.py

```diff
@@ -71,6 +80,66 @@ def _validate_fit_memory_usage(self, X_train):
             if model_memory_ratio > (0.20 * max_memory_usage_ratio):
                 raise NotEnoughMemoryError  # don't train full model to avoid OOM error
 
+    # TODO: Consider making this fully generic and available to all models
+    def _fit_with_samples(self, X_train, y_train, time_limit):
+        """
+        Fit model with samples of the data repeatedly, gradually increasing the amount of data until time_limit is reached or all data is used.
+
+        X_train and y_train must already be preprocessed
+        """
+        time_start = time.time()
+
+        sample_growth_factor = 2  # Growth factor of each sample in terms of row count
+        sample_time_growth_factor = 8  # Assume next sample will take 8x longer than previous (Somewhat safe but there are datasets where it is even >8x.
+
+        num_rows_samples = []
+        num_rows_max = len(X_train)
+        num_rows_cur = 10000
```

Comment:
Should `num_rows_cur` just be equal to all the data if `time_limits` is unspecified? Otherwise isn't the doubling strategy wasting unnecessary time?

Reply:
This is what occurs in _fit, this part of the code is only called when time_limits != None